spec/requests/users_controller_spec.rb

```diff
@@ -1517,11 +1517,26 @@ def post_user
           end
 
           context "custom_field" do
-            it "does not update the custom field" do
-              put "/u/#{user.username}.json", params: { custom_fields: { test: :it } }
+            before do
+              plugin = Plugin::Instance.new
+              plugin.register_editable_user_custom_field :test2
+            end
+
+            it "only updates allowed user fields" do
+              put "/u/#{user.username}.json", params: { custom_fields: { test1: :hello1, test2: :hello2 } }
+
+              expect(response.status).to eq(200)
```

Comment:
hmmm ... so the user send invalid params and we return a 200, I do not really agree with this ... we should return an 400 here.

Reply:
Fixing this will mean some changes in the client. 

At the moment every custom_field associated with the user is sent to the server when you save preferences. For example, if I hit save on my Meta profile, it sends up `custom_fields[staff_notes_count]: 1`. The server ignores this, so it is fine. If we return an error, it will break things.

The solution is add a 'editable user custom fields whitelist' to the client plugin API as well, but I wonder whether we should save that for after the 2.1 release?

Reply:
OK we can keep this "safe" change for 2.1 and do the proper fix for 2.2. We should not be asking the server to save fields it has no permission to save.